connectionchecker/connection_checker.go

```diff
@@ -89,17 +90,30 @@ func (cc *ConnectionChecker) sqlPing(ctx context.Context, db *sql.DB) error {
 	return err
 }
 
-func (cc *ConnectionChecker) checkMySQLConnection(ctx context.Context, dsn string) *agentpb.CheckConnectionResponse {
+func (cc *ConnectionChecker) checkMySQLConnection(ctx context.Context, dsn string, files *agentpb.TextFiles) *agentpb.CheckConnectionResponse {
 	var res agentpb.CheckConnectionResponse
+	var err error
 
-	cfg, err := mysql.ParseDSN(dsn)
+	if strings.Contains(dsn, "tls=custom") {
```

Comment:
For now, leave it as it is, but what if I set the password to `tls=custom`?
We need a better wat to detect if TLS is enabled.

Reply:
Fixed.